### PR TITLE
Create devcontainer.json

### PR DESCRIPTION
**Pull request by Sharpz7**
Fixes https://github.com/G-Research/armada/issues/1890

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-93) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by kannon92 — COMMENTED** [review]
Not sure what I am looking at.  What does this do? Are these images related to things you built for your codespaces work?

**Review by kannon92 — APPROVED** [review]

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# Install apt packages
+RUN add-apt-repository ppa:deadsnakes/ppa &&\
+    apt-get update &&\
+    apt-get install -y nano python3.9 &&\
+    # Cleanup
+    apt-get clean &&\
+    rm -rf /var/lib/apt/lists/*
+
+# Golang Install 1.19.4
```
Comment (kannon92) [inline .devcontainer/Dockerfile:19]: I don't know if this is a big deal but Armada is actually still on 1.18.

**Review by kannon92 — DISMISSED** [review]

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (Sharpz7) [reply .devcontainer/Dockerfile:19]: I think I install both 1.19 and 1.18 here?

I could maybe install a program to make it easier to switch between them... That would be a nice addition

Comment (kannon92) [reply .devcontainer/Dockerfile:19]: If it works then I’m happy. Just noticed it so feel free to ignore.

**Comment by Sharpz7** [comment]
This PR currently contains all the parts we would need for a demo codespace to work.

Two things:
- The resources are not big enough to actually run armada. Hopefully this changes in the future
- "Prebuilds ( https://docs.github.com/en/codespaces/prebuilding-your-codespaces/configuring-prebuilds#configuring-prebuilds) would allow for the setup to be skipped while staying devcontainer-native. I think this is 100% the route we want to go down.

I am happy for this to either stay as a PR or be merged in for use in the future.

**Comment by Sharpz7** [comment]
```bash
╭─coder@coder-sharpz7-testing ~/projects/armada/localdev  ‹devcontainer-testing› 
╰─➤  docker-compose logs executor
executor  | INFO[2023-02-28T21:10:27Z] Read base config from /app/config/executor/config.yaml 
executor  | INFO[2023-02-28T21:10:27Z] Read config from /app/config/executor/config.yaml 
executor  | INFO[2023-02-28T21:10:27Z] Starting http server listening on 9001       
executor  | INFO[2023-02-28T21:10:27Z] Creating pod checks...                       
executor  | INFO[2023-02-28T21:10:27Z]    Created event check Warning "Failed to pull image.*desc = failed to pull and unpack image" 1m30s Fail 
executor  | INFO[2023-02-28T21:10:27Z]    Created event check Warning "Failed to pull image.*code = Unknown desc = Error response from daemon" 1m30s Fail 
executor  | INFO[2023-02-28T21:10:27Z]    Created event check Warning NOT "nodes are available" 5m0s Retry 
executor  | INFO[2023-02-28T21:10:27Z]    Created container state check Waiting "InvalidImageName" 0s Fail 
executor  | INFO[2023-02-28T21:10:27Z]    Created container state check Waiting "ImagePullBackOff" 15m0s Retry 
executor  | INFO[2023-02-28T21:10:27Z]    Created container state check Waiting "ErrImagePull" 15m0s Retry 
executor  | INFO[2023-02-28T21:10:27Z]    Created container state check Waiting ".*" 5m0s Retry 
executor  | ERRO[2023-02-28T21:10:27Z] error receiving leases from server            error="rpc error: code = Unknown desc = [x-request-id: \"BKJhCgSD9oX4JXyNLDgAGV\"] no resources with non-zero weight available for scheduling on any cluster: resource scarcity map[], total resources {map[cpu:{{4 3} {<nil>} 4k DecimalSI} ephemeral-storage:{{0 0} {0xc000bb6510}  BinarySI} memory:{{68719476736000 0} {<nil>}  BinarySI}]}"
executor  | INFO[2023-02-28T21:10:27Z] Requesting new jobs with free resource cpu: 4000000m, memory 62.50Ti, ephemeral-storage: 125.00Ti. Received 0 new jobs.  
executor  | ERRO[2023-02-28T21:10:32Z] error receiving leases from server            error="rpc error: code = Unknown desc = [x-request-id: \"VaTPqKNSZCNnaW6L3qrd77\"] no resources with non-zero weight available for scheduling on any cluster: resource scarcity map[], total resources {map[cpu:{{4 3} {<nil>} 4k DecimalSI} ephemeral-storage:{{0 0} {0xc0006ff410}  BinarySI} memory:{{68719476736000 0} {<nil>}  BinarySI}]}"
```
```bash
╭─coder@coder-sharpz7-testing ~/projects/armada/localdev  ‹devcontainer-testing› 
╰─➤  docker-compose logs lookout 
lookout  | INFO[2023-02-28T21:10:27Z] Read base config from /app/config/lookout/config.yaml 
lookout  | INFO[2023-02-28T21:10:27Z] Read config from /app/config/lookout/config.yaml 
lookout  | INFO[2023-02-28T21:10:27Z] Updating database...                         
lookout  | INFO[2023-02-28T21:10:27Z] Current version 0                            
lookout  | panic: ERROR: duplicate key value violates unique constraint "pg_type_typname_nsp_index" (SQLSTATE 23505)
lookout  | 
lookout  | goroutine 1 [running]:
lookout  | main.main()
lookout  |      /go/src/armada/cmd/lookout/main.go:60 +0x788
```

```bash
Attaching to binoculars
binoculars  | INFO[2023-02-28T21:21:28Z] Read base config from /app/config/binoculars/config.yaml 
binoculars  | INFO[2023-02-28T21:21:28Z] Read config from /app/config/binoculars/config.yaml 
binoculars  | INFO[2023-02-28T21:21:28Z] Starting...                                  
binoculars  | ERRO[2023-02-28T21:21:28Z] Failed to connect to kubernetes because value %!q(float32=0) is invalid for field "qps": qps must be positive 
binoculars exited with code 255
```

I cannot figure out these errors.

**Pull request closed without merging**